docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,330 @@
+Configuration API Design
+===================
+
+## Motivation
+
+[Issue 51][issue_51] specifies the need to design an API for configuring Envoy Gateway at runtime. This configuration is
+referred to as the "Bootstrap Config" in the Envoy Gateway high-level [design doc][design_doc]. This document replaces
+the term "bootstrap" with "runtime" as some users may confuse bootstrap with Envoy's [bootstrap][envoy_boot]
```

Comment:
if replacing bootstrap with runtimeConfig, can you also update  https://github.com/envoyproxy/gateway/blob/main/docs/design/SYSTEM_DESIGN.md with this new terminology then
imho fine with either names

Reply:
I will like to handle the design doc update in a  separate PR since I need assistance with updating the diagram.

Reply:
https://github.com/envoyproxy/gateway/pull/111 updates the design doc and diagram accordingly.